### PR TITLE
Thought on tombstones

### PR DESCRIPTION
**Pull request by HuwCampbell**
Just writing up some of our thoughts from yesterday, and an idea on implementation (which is one extra construct).

### CODE REVIEW

**Comment by amosr** [comment]
this looks pretty excellent :moneybag:

**Comment by jystic** [comment]
:money_with_wings:

File: doc/design/tombstones.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+--------
+Any expression which `uses` a tombstone during its evaluation returns a missing value, however, a primitive will be provided to either
+filter or map tombstones, such that they can be handled explicitly.
+
+```
+feature x ~> filter (not tombstone) ~> count
+```
+will filter out all tombstone values and allow count to run.
+```
+feature x ~> let xx = if (tombstone) 0 else value ~> mean xx
```
Comment (markhibberd) [inline doc/design/tombstones.md]: Naive question here, do you want to be able to fold over these, or at least use context of previous value. For example, perhaps you want to be able keep the value the same as the last "known" value or similar.

**Comment by markhibberd** [comment]
:+1:

File: doc/design/tombstones.md (hunk 1 above)
Comment (amosr) [inline doc/design/tombstones.md]: yep, you'll be able to use this tombstone primitive inside a fold, so it should be fine

**Pull request merged**